CartingApp/Cart.cs

```diff
@@ -6,49 +6,71 @@ namespace CartingApp
 {
     public class Cart
     {
-        public List<CartItem> cartItemList;
-        public Bill bill;
+        public List<CartItem> cartItemList { get; private set; }
+        public Invoice invoice;
 
-        public Cart(DiscountType discountType)
+        public Cart(DiscountType discountType, double discount)
         {
-            bill = new Bill(discountType);
             cartItemList = new List<CartItem>();
+            invoice = new Invoice(this.cartItemList, DiscountType.CategoryDiscount);
         }
 
-        public void AddProduct(CartItem cartItem)
+        public void AddCartItem(CartItem cartItem)
         {
             cartItemList.Add(cartItem);
         }
 
-        public void RemoveProduct(string productName)
+        public void RemoveCartItem(string productName)
         {
-            cartItemList.Remove(
-                cartItemList.Find(x => x.product.name == productName)
+            try
```

Comment:
Check if product with the given name already exists or not and then throw custom exception.Do not suppress exceptions